others/iterative_tree_traversals.cpp

```diff
@@ -0,0 +1,362 @@
+ * @param root head/root node of a tree
+ * @return result that is containing the preorder traversal of a tree
+ */
+std::vector<int64_t> BinaryTree::preOrderIterative(Node *root) {   
+    std::stack<Node *> stack; ///< is used to find and traverse the child nodes.
+    std::vector<int64_t> result;  ///< list of values, sorted in pre-order.
+    
+    stack.push(root);
+    
+    while(!stack.empty()) {
+        result.push_back(stack.top()->data);
+        Node *current = stack.top();
+        stack.pop();
+        
+        if(current->right) {
+            stack.push(current->right);
+        }
+        if(current->left) {
+            stack.push(current->left);
+        }
+    }
+    
+    return result;
+}
+
+/**
+ * @brief postOrderIterative() function that will perform the postorder traversal iteratively,
+ * and return the result array that contain the postorder traversal of a tree.
+ * @param root head/root node of a tree
+ * @return result that is containing the postorder traversal of a tree
+ */
+std::vector<int64_t> BinaryTree::postOrderIterative(Node *root) {   
+    std::stack<Node *> stack; ///< is used to find and traverse the child nodes.
+    std::vector<int64_t> result;  ///< List of values, sorted in post-order.
+    
+    stack.push(root);
+    
+    while(!stack.empty()) {
+        result.push_back(stack.top()->data);
+        Node *current = stack.top();
+        stack.pop();
+        
+        if(current->left) {
+            stack.push(current->left);
+        }
+        if(current->right) {
+            stack.push(current->right);
+        }
+    }
+
+    reverse(result.begin(), result.end());
+    
+    return result;
+}
+
+/**
+ * @brief inOrderIterative() function that will perform the inorder traversal iteratively,
+ * and return the result array that contain the inorder traversal of a tree.
+ * @param root head/root node of a tree
+ * @return result that is containing the inorder traversal of a tree
+ */
+std::vector<int64_t> BinaryTree::inOrderIterative(Node *root) {
+    std::stack<Node *> stack; ///< is used to find and traverse the child nodes.
+    std::vector<int64_t> result;  ///< List of values, sorted in in-order.
+
+    Node *current = root;
+    
+    while(!stack.empty() || current) {
+        while(current) {
+            stack.push(current);
+            current = current->left;
+        }
+        current = stack.top();
+        stack.pop();    
+        result.push_back(current->data);   
+        current = current->right;
+    }   
+    return result;
+}
+}  //  namespace iterative_tree_traversals
+}  //  namespace others
+
+/**
+ * @brief Test the computed preorder with the actual preorder.
+ * @param binaryTree instance of BinaryTree class
+ * @param root head/root node of a tree
+ */
+static void test1(others::iterative_tree_traversals::BinaryTree binaryTree, others::iterative_tree_traversals::Node *root){
+    std::vector<int64_t> actual_result{1, 2, 4, 5, 3};

```

Comment:
Please add a one-line description of what the function is for (like the line below).

Reply:
A one-line description of what the function is doing for all the functions is already mentioned in the `@brief ` right?